compat/src/main/scala-2.11_2.12/scala/collection/compat/CompatImpl.scala

```diff
@@ -53,14 +51,31 @@ private final class IdentityPreservingBuilder[A, CC[X] <: TraversableOnce[X]](
   }
 
   def clear(): Unit = {
-    collection = null.asInstanceOf[CC[A]]
-    if (ruined) that.clear()
-    ruined = false
+    collection = null.asInstanceOf[C]
+    if (ruined) {
+      that.clear()
```

Comment:
is there also an invariant "`!ruined` => `that` is empty"?

Reply:
yes, elements are only added to `that` after `ruin()`